packages/SwingSet/src/kernel/vat-admin-hooks.js

```diff
@@ -27,38 +21,30 @@ export function makeVatAdminHooks(tools) {
         type: 'create-vat',
         vatID,
         source,
-        vatParameters,
+        vatParameters: marshalledVatParameters,
         dynamicOptions,
       };
       kernelKeeper.addToAcceptanceQueue(harden(event));
       // the device gets the new vatID immediately, and will be notified
       // later when it is created and a root object is available
-      const vatIDCapData = { body: JSON.stringify(vatID), slots: [] };
-      return harden(vatIDCapData);
+      return harden(kser(vatID));
     },
 
     createByID(argsCapData) {
       // argsCapData is marshal([bundleID, options]), and options is {
-      // vatParameters, ...rest }, and 'rest' is JSON-serializable (no
-      // slots or bigints or undefined). We get the intermediate marshal
-      // representation (with @qclass nodes), carve off vatParameters,
-      // then reassemble the rest. All slots will be associated with
-      // vatParameters.
+      // vatParameters, ...rest }, and 'rest' is JSON-serializable (no slots or
+      // bigints or undefined). We get the intermediate marshal representation,
+      // carve off vatParameters, then reassemble the rest. All slots will be
+      // associated with vatParameters.
```

Comment:
Now that you're using kunser on it instead of JSON.parse, is this comment still appropriate?

Reply:
As it turns out, yes.